versiondb/README.md

```diff
@@ -0,0 +1,141 @@
+
+## Migration
+
+Since our chain has pretty big now, a lot of efforts have been put to make sure the transition process can finish in practical time. The migration process will try to parallelize the tasks as much as possible, and use significant ram, but there's flags for user to control the concurrency level and ram usage to make it runnable on different machine specs.
+
+The legacy state migration process is done in two main steps:
+
+- Extract state change sets from existing archive IAVL tree.
+- Feed the change set files to versiondb.
+
+### Extract Change Sets
+
+```
+$ cronosd changeset dump --home /chain/.cronosd/ --output data distribution acc authz bank capability cronos evidence evm feegrant feeibc feemarket gov ibc mint params slashing staking transfer upgrade
+```
+
+`dump` command will extract the change sets from the IAVL tree, and store each store in separate directories. The change set files are segmented into different chunks and compressed with zlib level 6 by default, the chunk size defaults to 1m blocks, the result `data` directly will look like this:
+
+```
+data/acc/block-0.zz
+data/acc/block-1000000.zz
+data/acc/block-2000000.zz
+...
+data/authz/block-0.zz
+data/authz/block-1000000.zz
+data/authz/block-2000000.zz
+...
+```
+
+Extraction is the slowest step, the test run on testnet archive node takes around 11 hours on a 8core ssd machine, but fortunately, the change set files can be verified pretty fast, so they can be share on CDN in a trustless manner, normal users should just download them from CDN and verify the correctness locally, should be much faster than extract by yourself.
+
+#### Verify Change Sets
+
+```
+$ cronosd changeset verify data/acc/*.zz
+7130689 8DF52D6F7A7690916894AF67B07D64B678FB686626B2B3109813BBE172E74F08
+```
+
+`verify` command will replay all the change sets and rebuild the final IAVL tree and output the root hash of the final version, user can check the root hash against the IAVL tree.
+
+> TODO: will provide command to generate app-hash directly, combining the root hashes of each store, so it'll be easier for end user to verify.
+
+`verify` command takes several minutes and several gigabytes of ram to run, if ram usage is a problem, it can also run incrementally, you can export the snapshot for a middle version, then verify the remaining versions start from that snapshot:
+
+```
+$ cronosd changeset verify --save-snapshot /tmp/snapshot data/acc/block-0.zz data/acc/block-1000000.zz data/acc/block-2000000.zz
+$ cronosd changeset verify --load-snapshot /tmp/snapshot data/acc/block-3000000.zz data/acc/block-4000000.zz data/acc/block-5000000.zz
+```
```

Comment:
does the op/user requires to verify the changeset in each module manually? meaning verify it just assign the folder path then it will verify all files in the folder for better UX

Reply:
depends on if user trust the data provider.

Reply:
https://github.com/crypto-org-chain/cronos/issues/826
Opened a separate issue to improve cmd UX later.